### ISSUE TITLE
read_bvals_bvecs can't read a single volume dwi

### ISSUE THREAD

**Issue by soichih** [dipy/dipy#2046]
I have bvecs / bvals that looks like this (for the b0 data)

bvec
```
0
0
0
```

bval
```
0
```

When I try to open this with read_bvals_bvecs() it errors out with an error message like this.

```
from dipy.io import read_bvals_bvecs
bvals,bvecs = read_bvals_bvecs('bval', 'bvec')
```

```
Traceback (most recent call last):
  File "read_bvals_bvecs.py", line 2, in <module>
    bvals,bvecs = read_bvals_bvecs('bval', 'bvec')
  File "/home/hayashis/.local/lib/python2.7/site-packages/dipy/io/gradients.py", line 70, in read_bvals_bvecs
    raise IOError('bvec file should be saved as a two dimensional array')
IOError: bvec file should be saved as a two dimensional array
```

We are using dipy's read_bvals_bvecs to validate and normalize the bvec/bval, but then we run the actual processing using mrtrix3/fsl, etc.. that I believe can handle such b0 input file (I haven't tested). I believe read_bvals_bvecs should be able to parse single volume dwi - or at least just display warning instead of throwing exception?

**Comment by skoudoro** [comment]
Hi @soichih,

Thank you for asking! Indeed, When I look at [here](https://github.com/dipy/dipy/blob/master/dipy/io/gradients.py#L68), it seems forbidden but I do not know why.

I suppose there is a good reason, Can you explain to us @arokem @Garyfallidis or other?

**Comment by arokem** [comment]
This does seem like a bug to me. There's no reason a one-volume data-set
would be disallowed (although it is admittedly a rather strange
use-case...). It will require some special-casing and extra checking,
though. Do you want to take a stab at it, @soichih?

**Comment by soichih** [comment]
@arokem So instead of just removing this check, you want to check to see if the single volume bvecs contains some valid number? I use diyp.core.gradients.gradient_table() along with read_bvals_bvecs, and I believe https://github.com/dipy/dipy/blob/master/dipy/core/gradients.py#L114 provides enough checks already?

**Comment by arokem** [comment]
I think that the right thing to do here is to check whether we have a 2D array or an array with three numbers (which is this special case).

**Comment by soichih** [comment]
The line just above the 2D check (https://github.com/dipy/dipy/blob/master/dipy/io/gradients.py#L65)  ensures that it's an array with 3 numbers, right?

**Comment by arokem** [comment]
Yeah - you're right. Maybe we should downgrade the error on L67 to a
warning? I think that most users will want to know if they're data looks
like it's a single b-vec, when it  shouldn't be.

**Comment by soichih** [comment]
@bacaron just ran into this issue. 

Maybe it could also check the bvals and if they are near-0, then assume that the image is b0-only image and only show warnings?

**Comment by skoudoro** [comment]
I will create a PR to fix that @soichih, I plan to allow it and just add a warning to the user as proposed above.

So this feature will be available for the next release (1.3), next month.
